<a href="https://colab.research.google.com/github/aaryav13/aaryav13/blob/main/BellaBeat_Case_Study_with_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# BellaBeat- A Wellness Technology Company

Founded in 2013 by * Urška Sršen* and *Sandro Mur*, **BellaBeat** is a high-tech manufacturer of health-focused wearables gadgets for women.By 2016, BellaBeat had opened numbers of branches around the world and launched multiple products like, Leaf, Time, Spring and so on. The aesthetically pleasing designs based products collect data activity, sleep, stress, and reproductive health to empower women with the knowledge about their own health and habits.

## Business Task

- Analyze smart device usage data in order to gain insight into how consumers use non-Bellabeat smart
devices.

## Questions 
- What are the trends in the non-Bellabeat devices usages?
- What are the insights gain from these trends ?
- How could these insights guide BellaBeat marketing strategy?

## Required Libraries


In [1]:

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
from datetime import datetime as dt

In [2]:
import plotly.express as px


## Downloading Date

This [date](https://www.kaggle.com/arashnic/fitbit) set is taken from Kaggle website.Thirty eligible Fitbit users consented to the submission of
personal tracker data, which includes their minute-level output for physical activity, heart rate, and sleep monitoring. It also includes
information about daily activity, steps, and heart rate that can be used to explore users’ habits.

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
# Data Loading

sleep_df = pd.read_csv("/content/drive/MyDrive/FitBit Fitness Tracker Data.zip (Unzipped Files)/Fitabase Data 4.12.16-5.12.16/sleepDay_merged.csv")
calories_df = pd.read_csv("/content/drive/MyDrive/FitBit Fitness Tracker Data.zip (Unzipped Files)/Fitabase Data 4.12.16-5.12.16/hourlyCalories_merged.csv")
intensities_df = pd.read_csv("/content/drive/MyDrive/FitBit Fitness Tracker Data.zip (Unzipped Files)/Fitabase Data 4.12.16-5.12.16/hourlyIntensities_merged.csv")
activity_df = pd.read_csv("/content/drive/MyDrive/FitBit Fitness Tracker Data.zip (Unzipped Files)/Fitabase Data 4.12.16-5.12.16/dailyActivity_merged.csv")
weight_df = pd.read_csv("/content/drive/MyDrive/FitBit Fitness Tracker Data.zip (Unzipped Files)/Fitabase Data 4.12.16-5.12.16/weightLogInfo_merged.csv")

## Datesets-At-Glance

This will give information about the datasets using Head() function.

In [5]:
sleep_df.head()

Id               SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
0  1503960366  4/12/2016 12:00:00 AM                  1                 327   
1  1503960366  4/13/2016 12:00:00 AM                  2                 384   
2  1503960366  4/15/2016 12:00:00 AM                  1                 412   
3  1503960366  4/16/2016 12:00:00 AM                  2                 340   
4  1503960366  4/17/2016 12:00:00 AM                  1                 700   

   TotalTimeInBed  
0             346  
1             407  
2             442  
3             367  
4             712

In [6]:
calories_df.head()

Id           ActivityHour  Calories
0  1503960366  4/12/2016 12:00:00 AM        81
1  1503960366   4/12/2016 1:00:00 AM        61
2  1503960366   4/12/2016 2:00:00 AM        59
3  1503960366   4/12/2016 3:00:00 AM        47
4  1503960366   4/12/2016 4:00:00 AM        48

In [7]:
intensities_df.head()

Id           ActivityHour  TotalIntensity  AverageIntensity
0  1503960366  4/12/2016 12:00:00 AM              20          0.333333
1  1503960366   4/12/2016 1:00:00 AM               8          0.133333
2  1503960366   4/12/2016 2:00:00 AM               7          0.116667
3  1503960366   4/12/2016 3:00:00 AM               0          0.000000
4  1503960366   4/12/2016 4:00:00 AM               0          0.000000

In [8]:
activity_df.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    4/12/2016       13162           8.50             8.50   
1  1503960366    4/13/2016       10735           6.97             6.97   
2  1503960366    4/14/2016       10460           6.74             6.74   
3  1503960366    4/15/2016        9762           6.28             6.28   
4  1503960366    4/16/2016       12669           8.16             8.16   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.57                      0.69   
2                       0.0                2.44                      0.40   
3                       0.0                2.14                      1.26   
4                       0.0                2.71                      0.41   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 4.71                      0.0                 21   
2                 3.91                      0.0                 30   
3                 2.83                      0.0                 29   
4                 5.04                      0.0                 36   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                   13                   328               728      1985  
1                   19                   217               776      1797  
2                   11                   181              1218      1776  
3                   34                   209               726      1745  
4                   10                   221               773      1863

In [9]:
weight_df.head()

Id                   Date    WeightKg  WeightPounds   Fat  \
0  1503960366   5/2/2016 11:59:59 PM   52.599998    115.963147  22.0   
1  1503960366   5/3/2016 11:59:59 PM   52.599998    115.963147   NaN   
2  1927972279   4/13/2016 1:08:52 AM  133.500000    294.317120   NaN   
3  2873212765  4/21/2016 11:59:59 PM   56.700001    125.002104   NaN   
4  2873212765  5/12/2016 11:59:59 PM   57.299999    126.324875   NaN   

         BMI  IsManualReport          LogId  
0  22.650000            True  1462233599000  
1  22.650000            True  1462319999000  
2  47.540001           False  1460509732000  
3  21.450001            True  1461283199000  
4  21.690001            True  1463097599000

Having consistent date format in all the datasets will us further in the analysis.

## Consistent Date Formats

In [10]:
## Changing the date format

# Sleep
sleep_df['SleepDay'] = pd.to_datetime(sleep_df['SleepDay'] )

sleep_df['new_date'] = [d.date() for d in sleep_df['SleepDay']]


# Calories
calories_df['ActivityHour'] = pd.to_datetime(calories_df['ActivityHour'])

calories_df['new_date'] = [d.date() for d in calories_df['ActivityHour']]
calories_df['new_time'] = [d.time() for d in calories_df['ActivityHour']]


# Intensities
intensities_df['ActivityHour'] = pd.to_datetime(intensities_df['ActivityHour'])

intensities_df['new_date'] = [d.date() for d in intensities_df['ActivityHour']]
intensities_df['new_time'] = [d.time() for d in intensities_df['ActivityHour']]


# Activity
activity_df['ActivityDate'] = pd.to_datetime(activity_df['ActivityDate'])

activity_df['new_date'] = [d.date() for d in activity_df['ActivityDate']]


# Weight
weight_df['Date'] = pd.to_datetime(weight_df['Date'])

weight_df['new_date'] = [d.date() for d in weight_df['Date']]
weight_df['new_time'] = [d.time() for d in weight_df['Date']]


## Uniques Values & Summary of Data

In [11]:
# Unique values in each dataframes

# Sleep

sleep_unique = sleep_df['Id'].nunique()
print(f'sleep_df :{sleep_unique}')
print('   ')

# Calories

calories_unique = calories_df['Id'].nunique()
print( f'calories_df : {calories_unique}')

# Intensities

intensities_unique = intensities_df['Id'].nunique()
print('')
print( f'intensities_df : {intensities_unique}')

# Acitvity

activity_unique = activity_df['Id'].nunique()
print('')
print( f'activity_df : {activity_unique}')

# Weight

weight_unique = weight_df['Id'].nunique()
print('')
print(f'weight_df : {weight_unique}')

sleep_df :24
   
calories_df : 33

intensities_df : 33

activity_df : 33

weight_df : 8


Here, Calories, activity and intensities dataset have 33 distinct values but sleep and weight data is provided by only 24 and 8 users respectively.

**Limitation** : Less number of participants in weight dataset would not provide impactful insights.

### Summary of each dataframe

In [12]:
## Sleep
sleep_df.describe().drop(columns =['Id'])

TotalSleepRecords  TotalMinutesAsleep  TotalTimeInBed
count         413.000000          413.000000      413.000000
mean            1.118644          419.467312      458.639225
std             0.345521          118.344679      127.101607
min             1.000000           58.000000       61.000000
25%             1.000000          361.000000      403.000000
50%             1.000000          433.000000      463.000000
75%             1.000000          490.000000      526.000000
max             3.000000          796.000000      961.000000

In [13]:
## Calories
calories_df.describe().drop(columns =['Id'])

Calories
count  22099.000000
mean      97.386760
std       60.702622
min       42.000000
25%       63.000000
50%       83.000000
75%      108.000000
max      948.000000

In [14]:
## Activity
activity_df.describe().drop(columns =['Id'])

TotalSteps  TotalDistance  TrackerDistance  LoggedActivitiesDistance  \
count    940.000000     940.000000       940.000000                940.000000   
mean    7637.910638       5.489702         5.475351                  0.108171   
std     5087.150742       3.924606         3.907276                  0.619897   
min        0.000000       0.000000         0.000000                  0.000000   
25%     3789.750000       2.620000         2.620000                  0.000000   
50%     7405.500000       5.245000         5.245000                  0.000000   
75%    10727.000000       7.712500         7.710000                  0.000000   
max    36019.000000      28.030001        28.030001                  4.942142   

       VeryActiveDistance  ModeratelyActiveDistance  LightActiveDistance  \
count          940.000000                940.000000           940.000000   
mean             1.502681                  0.567543             3.340819   
std              2.658941                  0.883580             2.040655   
min              0.000000                  0.000000             0.000000   
25%              0.000000                  0.000000             1.945000   
50%              0.210000                  0.240000             3.365000   
75%              2.052500                  0.800000             4.782500   
max             21.920000                  6.480000            10.710000   

       SedentaryActiveDistance  VeryActiveMinutes  FairlyActiveMinutes  \
count               940.000000         940.000000           940.000000   
mean                  0.001606          21.164894            13.564894   
std                   0.007346          32.844803            19.987404   
min                   0.000000           0.000000             0.000000   
25%                   0.000000           0.000000             0.000000   
50%                   0.000000           4.000000             6.000000   
75%                   0.000000          32.000000            19.000000   
max                   0.110000         210.000000           143.000000   

       LightlyActiveMinutes  SedentaryMinutes     Calories  
count            940.000000        940.000000   940.000000  
mean             192.812766        991.210638  2303.609574  
std              109.174700        301.267437   718.166862  
min                0.000000          0.000000     0.000000  
25%              127.000000        729.750000  1828.500000  
50%              199.000000       1057.500000  2134.000000  
75%              264.000000       1229.500000  2793.250000  
max              518.000000       1440.000000  4900.000000

In [15]:
## Weight

weight_df.describe().drop(columns =['Id','LogId'])

WeightKg  WeightPounds       Fat        BMI
count   67.000000     67.000000   2.00000  67.000000
mean    72.035821    158.811801  23.50000  25.185224
std     13.923206     30.695415   2.12132   3.066963
min     52.599998    115.963147  22.00000  21.450001
25%     61.400002    135.363832  22.75000  23.959999
50%     62.500000    137.788914  23.50000  24.389999
75%     85.049999    187.503152  24.25000  25.559999
max    133.500000    294.317120  25.00000  47.540001

#### Observation

- Minimum minutes for sleep is '58' minutes and average sleep time is approximately 419 minutes which is around 7 hours of sleep. People needs 7-8 hours of [sleep](https://www.webmd.com/sleep-disorders/sleep-requirements#:~:text=Most%20adults%20need%207%20to,hours%20of%20sleep%20than%20usual.) and 75% of people get this much sleep.

- The average amount of calories burn hourly is approx. 97 calories.

- The average of SedentaryMinutes is approx. 991 that is 16 hours, which is slighlty higher than of an [office worker](https://www.healthline.com/nutrition/why-sitting-is-bad-for-you#weight-gain) and bad for health.

- On average, one person takes 7638 steps in a day which is slightly less than Average (somewhat active) catogory which ranges from 7,500 to 9,999 steps per day.CDC recommend that most adults aim for 10,000 steps per dayTrusted Source. 

## Combining data

Merging two data sets for better analysis of sleep hours and acitivities  per day on Date column.

In [16]:
# merging sleep and activites data

combined_df = pd.merge(left =sleep_df, right= activity_df 
                            ,how="inner",on= ['Id','new_date']).drop(columns=['SleepDay','ActivityDate'])

In [17]:
combined_df

Id  TotalSleepRecords  TotalMinutesAsleep  TotalTimeInBed  \
0    1503960366                  1                 327             346   
1    1503960366                  2                 384             407   
2    1503960366                  1                 412             442   
3    1503960366                  2                 340             367   
4    1503960366                  1                 700             712   
..          ...                ...                 ...             ...   
408  8792009665                  1                 343             360   
409  8792009665                  1                 503             527   
410  8792009665                  1                 415             423   
411  8792009665                  1                 516             545   
412  8792009665                  1                 439             463   

       new_date  TotalSteps  TotalDistance  TrackerDistance  \
0    2016-04-12       13162           8.50             8.50   
1    2016-04-13       10735           6.97             6.97   
2    2016-04-15        9762           6.28             6.28   
3    2016-04-16       12669           8.16             8.16   
4    2016-04-17        9705           6.48             6.48   
..          ...         ...            ...              ...   
408  2016-04-30        7174           4.59             4.59   
409  2016-05-01        1619           1.04             1.04   
410  2016-05-02        1831           1.17             1.17   
411  2016-05-03        2421           1.55             1.55   
412  2016-05-04        2283           1.46             1.46   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                         0.0                1.88                      0.55   
1                         0.0                1.57                      0.69   
2                         0.0                2.14                      1.26   
3                         0.0                2.71                      0.41   
4                         0.0                3.19                      0.78   
..                        ...                 ...                       ...   
408                       0.0                0.33                      0.36   
409                       0.0                0.00                      0.00   
410                       0.0                0.00                      0.00   
411                       0.0                0.00                      0.00   
412                       0.0                0.00                      0.00   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                   6.06                      0.0                 25   
1                   4.71                      0.0                 21   
2                   2.83                      0.0                 29   
3                   5.04                      0.0                 36   
4                   2.51                      0.0                 38   
..                   ...                      ...                ...   
408                 3.91                      0.0                 10   
409                 1.04                      0.0                  0   
410                 1.17                      0.0                  0   
411                 1.55                      0.0                  0   
412                 1.46                      0.0                  0   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                     13                   328               728      1985  
1                     19                   217               776      1797  
2                     34                   209               726      1745  
3                     10                   221               773      1863  
4                     20                   164               539      1728  
..                   ...                   ...               ...       ...  
408           

In [18]:
combined_df.head()

Id  TotalSleepRecords  TotalMinutesAsleep  TotalTimeInBed  \
0  1503960366                  1                 327             346   
1  1503960366                  2                 384             407   
2  1503960366                  1                 412             442   
3  1503960366                  2                 340             367   
4  1503960366                  1                 700             712   

     new_date  TotalSteps  TotalDistance  TrackerDistance  \
0  2016-04-12       13162           8.50             8.50   
1  2016-04-13       10735           6.97             6.97   
2  2016-04-15        9762           6.28             6.28   
3  2016-04-16       12669           8.16             8.16   
4  2016-04-17        9705           6.48             6.48   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.57                      0.69   
2                       0.0                2.14                      1.26   
3                       0.0                2.71                      0.41   
4                       0.0                3.19                      0.78   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 4.71                      0.0                 21   
2                 2.83                      0.0                 29   
3                 5.04                      0.0                 36   
4                 2.51                      0.0                 38   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                   13                   328               728      1985  
1                   19                   217               776      1797  
2                   34                   209               726      1745  
3                   10                   221               773      1863  
4                   20                   164               539      1728

In [42]:
# Combined Columns
combined_df.columns

Index(['Id', 'TotalSleepRecords', 'TotalMinutesAsleep', 'TotalTimeInBed',
       'new_date', 'TotalSteps', 'TotalDistance', 'TrackerDistance',
       'LoggedActivitiesDistance', 'VeryActiveDistance',
       'ModeratelyActiveDistance', 'LightActiveDistance',
       'SedentaryActiveDistance', 'VeryActiveMinutes', 'FairlyActiveMinutes',
       'LightlyActiveMinutes', 'SedentaryMinutes', 'Calories'],
      dtype='object')

###  Data Visualization

In [47]:
fig = px.scatter(combined_df, x="TotalSteps", y="Calories",width=700, height=600, trendline='ols',trendline_scope="overall")
fig.update_layout(title= 'Total Steps vs. Calories', 
                  xaxis_title="Total Steps", yaxis_title="Calories")
fig.show();

Here we can see a fairly positive correaltion between "TotalSteps" and "Calories". 

In [44]:
fig = px.scatter(combined_df, x="TotalMinutesAsleep", y="TotalTimeInBed",width=700, height=600, trendline='ols',trendline_scope="overall")
fig.show()

Here, we can see a srong positive corrrelation between the amount of time spent in bed and the minutes of sleep.

In [22]:
intensities_df

Id        ActivityHour  TotalIntensity  AverageIntensity  \
0      1503960366 2016-04-12 00:00:00              20          0.333333   
1      1503960366 2016-04-12 01:00:00               8          0.133333   
2      1503960366 2016-04-12 02:00:00               7          0.116667   
3      1503960366 2016-04-12 03:00:00               0          0.000000   
4      1503960366 2016-04-12 04:00:00               0          0.000000   
...           ...                 ...             ...               ...   
22094  8877689391 2016-05-12 10:00:00              12          0.200000   
22095  8877689391 2016-05-12 11:00:00              29          0.483333   
22096  8877689391 2016-05-12 12:00:00              93          1.550000   
22097  8877689391 2016-05-12 13:00:00               6          0.100000   
22098  8877689391 2016-05-12 14:00:00               9          0.150000   

         new_date  new_time  
0      2016-04-12  00:00:00  
1      2016-04-12  01:00:00  
2      2016-04-12  02:00:00  
3      2016-04-12  03:00:00  
4      2016-04-12  04:00:00  
...           ...       ...  
22094  2016-05-12  10:00:00  
22095  2016-05-12  11:00:00  
22096  2016-05-12  12:00:00  
22097  2016-05-12  13:00:00  
22098  2016-05-12  14:00:00  

[22099 rows x 6 columns]

In [23]:
intensities_new = intensities_df.groupby(by='new_time', dropna=True,as_index=False)

In [24]:
intensities_new = intensities_df.groupby(by='new_time', dropna=True,as_index=False).mean()

In [59]:
# Hourly Intensities
intensities_new

new_time            Id  TotalIntensity  AverageIntensity
0   00:00:00  4.846554e+09        2.129550          0.035493
1   01:00:00  4.844238e+09        1.419078          0.023651
2   02:00:00  4.844238e+09        1.043944          0.017399
3   03:00:00  4.844238e+09        0.443730          0.007396
4   04:00:00  4.844801e+09        0.633047          0.010551
5   05:00:00  4.844801e+09        4.950644          0.082511
6   06:00:00  4.847483e+09        7.771214          0.129520
7   07:00:00  4.847483e+09       10.733620          0.178894
8   08:00:00  4.847483e+09       14.668099          0.244468
9   09:00:00  4.847483e+09       15.387755          0.256463
10  10:00:00  4.844647e+09       17.643703          0.294062
11  11:00:00  4.848036e+09       16.921251          0.282021
12  12:00:00  4.847948e+09       19.847072          0.330785
13  13:00:00  4.845518e+09       18.775244          0.312921
14  14:00:00  4.845518e+09       18.868621          0.314477
15  15:00:00  4.837441e+09       15.584699          0.259745
16  16:00:00  4.855342e+09       17.716648          0.295277
17  17:00:00  4.853758e+09       21.655629          0.360927
18  18:00:00  4.853758e+09       21.921634          0.365361
19  19:00:00  4.853758e+09       21.385210          0.356420
20  20:00:00  4.853758e+09       14.339956          0.238999
21  21:00:00  4.857459e+09       12.072928          0.201215
22  22:00:00  4.853337e+09        9.063053          0.151051
23  23:00:00  4.849572e+09        4.996678          0.083278

In [45]:
fig = px.histogram(intensities_new, y='AverageIntensity',x='new_time',width=700, height=600)
fig.update_layout(title= 'Time vs. Average Total Intensity', 
                  xaxis_title="Time", yaxis_title="Average total intensity")
fig.show()


- Users are highly engage in activiy between 5pm - 8 pm.
- Least active between 11 pm - 5 am.

In [48]:
fig = px.scatter(combined_df, x="TotalMinutesAsleep", y="SedentaryMinutes",width=700, height=600, trendline='lowess',trendline_scope="overall")
fig.update_layout(title= 'Total Minutes Asleep vs. Sedentary Minutes')
fig.show()

Total minutes asleep and Sedentary minutes have a mildly negative correlation, which means users might be able to increase their sleep minutes by reducing 'sedentary minutes' everyday.

### Recommendations

Insights after Analysing FitBit Data

- To reduce **Sedentary** minutes, **BellaBeat** can notify its users between every 2-3 hours that will remind the users to be active in between.

- BellaBeat App can give more recommendation for how to reduce time spend in bed and get sound sleep of 7-8 hours like, provide few relaxing meditation tutorial around 11 pm.

- Since users are moslty active around 5pm - 8 pm, BellaBeat app can even let users to share there stories with other and motivate each other users to get some exerise done during this time.

- At least 50 % people are slightly overweight in the provided data of 8 users.**Need more data for further analysis on BMI **.